src/box/gc.c

```diff
@@ -594,8 +607,10 @@ gc_checkpoint_fiber_f(va_list ap)
 			/* Periodic checkpointing is disabled. */
 			timeout = TIMEOUT_INFINITY;
 		}
-		if (!fiber_yield_timeout(timeout) &&
-		    !gc.checkpoint_is_pending) {
+		bool timed_out = fiber_yield_timeout(timeout);
+		if (fiber_is_cancelled())
+			break;
+		if (!timed_out && !gc.checkpoint_is_pending) {
```

Comment:
AFAIU `gc_do_checkpoint` may yield and then return without checking if the fiber was cancelled. In this case, we'd continue to the next iteration and yielded for the timeout. Better leave the check in the `while` condition.

Reply:
Fixed.

```diff
diff --git a/src/box/gc.c b/src/box/gc.c
index b3e2e3c1d..41da69179 100644
--- a/src/box/gc.c
+++ b/src/box/gc.c
@@ -592,7 +592,7 @@ gc_checkpoint_fiber_f(va_list ap)
        (void)ap;

        struct checkpoint_schedule *sched = &gc.checkpoint_schedule;
-       while (true) {
+       while (!fiber_is_cancelled()) {
                fiber_check_gc();
                double timeout = checkpoint_schedule_timeout(sched,
                                        ev_monotonic_now(loop()));
```